rundeckapp/grails-spa/packages/ui-trellis/src/library/components/storage/tests/KeyStorageEdit.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { mount } from "@vue/test-utils";
+import KeyStorageEdit from "../KeyStorageEdit.vue";
+
+const mountKeyStorageEdit = async (props = {}) => {
+  return mount(KeyStorageEdit, {
+    props: {
+      storageFilter: "",
+      uploadSetting: {
+        modifyMode: false,
+        keyType: "privateKey",
+        inputPath: "",
+        inputType: "text",
+        fileName: null,
+        file: null,
+        fileContent: "",
+        textArea: "some-text",
+        password: "",
+        status: "new",
+        errorMsg: null,
+        dontOverwrite: false,
+      },
+      project: "test-project",
+      ...props,
+    },
+  });
+};
+describe("KeyStorageEdit", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("emits cancelEditing when cancel button is clicked", async () => {
+    const wrapper = await mountKeyStorageEdit();
+    const cancelButton = wrapper.find('[data-testid="cancel-btn"]');
+    await cancelButton.trigger("click");
+    expect(wrapper.emitted("cancelEditing")).toHaveLength(1);
+    expect(Object.keys(wrapper.emitted())).toContain("cancelEditing");
+  });
+
+  it("emits finishEditing with correct data when Save button is clicked", async () => {
+    const wrapper = await mountKeyStorageEdit();
+    const result = {
+      success: true,
+      keyDetails: {
+        name: "exampleKey",
+        path: "/keys/exampleKey",
+        keyType: "privateKey",
+      },
+    };
+    jest.spyOn(wrapper.vm as any, "handleUploadKey").mockImplementation(() => {
+      wrapper.vm.$emit("finishEditing", result);
+    });
```

Comment:
blocker: Unit tests shouldn't override the implementation of the tested component. It's fine to override the implementation of dependencies (ie: rundeckContext.rundeckClient.storageKeyGetMaterial) because the dependency should have its own tests, but by overriding the implementation of `handleUploadKey`, the test won't be useful to catch any future issues. 

Also, would it be possible to add tests covering what will happen when the save button is clicked and:
- uploadSetting.keyType is password;
- uploadSetting.keyType is publicKey;
- exists (const inside of handleUploadKey) is false;
- exists is true;